src/components/MAPI/organizations/OrganizationDetailGeneral/utils.ts

```diff
@@ -410,3 +416,64 @@ export function fetchReleasesSummaryKey(
 
   return `fetchReleasesSummary/${clusterKeys.join()}`;
 }
+
+/**
+ * Get various statistics about the apps installed in the given clusters.
+ * @param httpClientFactory
+ * @param auth
+ * @param clusters
+ */
+export async function fetchAppsSummary(
+  httpClientFactory: HttpClientFactory,
+  auth: IOAuth2Provider,
+  clusters: capiv1alpha3.ICluster[]
+): Promise<ui.IOrganizationDetailAppsSummary> {
+  const summary: ui.IOrganizationDetailAppsSummary = {};
+
+  // The key is the app name, and the value is the number of deployments.
+  const apps: Record<string, number> = {};
+
+  try {
+    // Get all apps that belong to all clusters.
+    const appLists = await Promise.all(
+      clusters.map((cluster) => {
+        return applicationv1alpha1.getAppList(httpClientFactory(), auth, {
+          namespace: cluster.metadata.name,
+        });
+      })
+    );
+
+    for (const appList of appLists) {
+      for (const app of appList.items) {
+        apps[app.spec.name] ??= 0;
+        apps[app.spec.name] += 1;
+      }
+    }
+  } catch (err) {
+    ErrorReporter.getInstance().notify(err);
+  }
+
+  summary.appsInUseCount = Object.keys(apps).length;
+  summary.appDeploymentsCount = Object.values(apps).reduce(
+    (acc, curr) => acc + curr,
+    0
+  );
+
+  return summary;
+}
+
+/**
+ * The key used for caching the releases summary.
+ * @param clusters
+ */
+export function fetchAppsSummaryKey(
+  clusters?: capiv1alpha3.ICluster[]
+): string | null {
+  if (!clusters) return null;
+
+  const clusterKeys = clusters.map(
+    (c) => `${c.metadata.namespace}/${c.metadata.name}`
+  );
```

Comment:
Shouldn't clusters be sorted first?

Reply:
That would be the intelligent thing to do 😬